### PR TITLE
 Remove deprecated config parameters from near cache config

### PR DESCRIPTION
**Pull request by ahmetmircik**
ee counterpart: https://github.com/hazelcast/hazelcast-enterprise/pull/3082

### CODE REVIEW

File: hazelcast-client/src/main/resources/hazelcast-client-full-example.yaml [hunk 1]
```diff
@@ -588,7 +585,6 @@ hazelcast-client:
       max-idle-seconds: 100
       eviction-policy: LFU
       invalidate-on-change: true
```
Comment (mmedenjak) [inline hazelcast-client/src/main/resources/hazelcast-client-full-example.yaml]: You can remove the `max-size: 2000` and `eviction-policy: LFU` here.

**Comment by mmedenjak** [comment]
You can also remove the settings in `/Users/mmedenjak/workspace/checkouts/hazelcast/hazelcast/src/main/resources/hazelcast-full-example.yaml`

File: hazelcast-client/src/main/resources/hazelcast-client-full-example.yaml (hunk 1 above)
Comment (ahmetmircik) [reply hazelcast-client/src/main/resources/hazelcast-client-full-example.yaml]: done.

File: hazelcast/src/main/java/com/hazelcast/config/EvictionConfig.java [hunk 2]
```diff
@@ -111,42 +119,13 @@
...
         this.size = checkPositive(config.size, "Size must be positive number!");
         this.maxSizePolicy = checkNotNull(config.maxSizePolicy, "Max-Size policy cannot be null!");
-        if (config.evictionPolicy != null) {
-            this.evictionPolicy = config.evictionPolicy;
-        }
-        // Eviction policy comparator class name is not allowed to be null
-        if (config.comparatorClassName != null) {
-            this.comparatorClassName = config.comparatorClassName;
-        }
-        // Eviction policy comparator is not allowed to be null
```
[the remark this answers is not in the document]
Comment (ahmetmircik) [reply hazelcast/src/main/java/com/hazelcast/config/EvictionConfig.java]: they can't be set to null. Setter methods check nullability. I deleted other constructors due to the complexity they bring.

File: hazelcast/src/main/java/com/hazelcast/config/EvictionConfig.java [hunk 3]
```diff
@@ -32,16 +31,24 @@
 import static com.hazelcast.util.Preconditions.checkNotNull;
 import static com.hazelcast.util.Preconditions.checkPositive;
 
+//TODO how to translate when size is zero edit docs. Is it Integer.MAX
```
[the remark this answers is not in the document]
Comment (ahmetmircik) [reply hazelcast/src/main/java/com/hazelcast/config/EvictionConfig.java]: removed it.

File: hazelcast/src/main/java/com/hazelcast/config/EvictionConfig.java [hunk 4]
```diff
@@ -28,24 +27,33 @@
...
  * <p>
  * The default values of the eviction configuration are
  * <ul>
- * <li>{@link EvictionPolicy#LRU} as eviction policy</li>
- * <li>{@link EvictionConfig.MaxSizePolicy#ENTRY_COUNT} as max size policy</li>
- * <li>{@value DEFAULT_MAX_ENTRY_COUNT_FOR_ON_HEAP_MAP} as maximum size for on-heap {@link IMap}</li>
- * <li>{@value DEFAULT_MAX_ENTRY_COUNT} as maximum size for all other data structures and configurations</li>
+ * <li>
+ * {@link EvictionPolicy#LRU} as eviction policy
+ * </li> <li>{@link
```
Comment (vojtechtoman) [inline hazelcast/src/main/java/com/hazelcast/config/EvictionConfig.java]: Questionable reformatting.

Comment (ahmetmircik) [reply hazelcast/src/main/java/com/hazelcast/config/EvictionConfig.java]: reverted.

**Pull request merged**